core_backend/app/users/models.py

```diff
@@ -0,0 +1,88 @@
+from datetime import datetime
+from typing import Optional
+
+from sqlalchemy import (
+    DateTime,
+    String,
+    # delete,
+    select,
+)
+from sqlalchemy.exc import NoResultFound
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.orm import Mapped, mapped_column
+
+from ..models import Base
+from .schemas import UserCreate
+
+
+class UserDB(Base):
+    """
+    SQL Alchemy data model for users
+    """
+
+    __tablename__ = "user"
+
+    user_id: Mapped[str] = mapped_column(String, primary_key=True, nullable=False)
+    username: Mapped[str] = mapped_column(String, nullable=False)
+
+    # retrieval_token: Mapped[str] = mapped_column(String, nullable=False)
+
+    created_datetime_utc: Mapped[datetime] = mapped_column(DateTime, nullable=False)
+    updated_datetime_utc: Mapped[datetime] = mapped_column(DateTime, nullable=False)
```

Comment:
What would an update look like?

Reply:
Suggest scrapping `updated_datetime_utc` for now if there is no use

Reply:
Or maybe we want `last_active` or something similar

Reply:
I added "last_active" as a todo for later - it might come in handy. For now I haven't implemented any updating mechanisms, but PR #179 adds an endpoint for updating the retrieval_key (which will be a column in this table) and then we use the `updated_datetime_utc` column.